src/System.Private.ServiceModel/tools/scripts/StartWCFService.cmd

```diff
@@ -0,0 +1,53 @@
+@echo off 
+setlocal
+
+REM 
+REM Start WCF Self Hosted Service on local machine
+REM
+
+REM Build tools
+call BuildWCFSelfHostedService.cmd
+if NOT [%ERRORLEVEL%]==[0] (
+    echo ERROR: An error occurred while building WCF Self hosted Service.
+    set __EXITCODE=%ERRORLEVEL% 
+    goto :done
+  )
+call BuildCertUtil.cmd
+if NOT [%ERRORLEVEL%]==[0] (
+    echo ERROR: An error occurred while building the Certificate generator.
+    set __EXITCODE=%ERRORLEVEL%
+    goto :done
+  )
+
+
+REM Config Certs
+REM we need the direcotry to save the test.crl file. We are investigate a way to get rid of it
+REM
+md c:\wcftest
+pushd ..\..\..\..\bin\Wcf\tools\CertificateGenerator
+call CertificateGenerator.exe
+set __EXITCODE=%ERRORLEVEL%
+popd
+if NOT [%__EXITCODE%]==[0] (
+    echo Warning: An error occurred while running certificate generator.
+  )
+
+REM skip the checking of error code as OpenFirewallPorts can return different error code even it's successful
+call OpenFirewallPorts.cmd
+
+powershell -NoProfile -ExecutionPolicy unrestricted .\ConfigHttpsPort.ps1
+if NOT [%ERRORLEVEL%]==[0] (
+    echo Warning: An error occurred while configuration https port.
+  )
```

Comment:
IMHO, I'd prefer to log everything that goes wrong to a known log file and keep going.  At the end, check that "anything" went wrong and issue an ECHO about it.   This is how clean.cmd works, and it seems to be a good compromise.